base/linalg/triangular.jl

```diff
@@ -1894,15 +1884,20 @@ function sqrtm{T,realmatrix}(A::UpperTriangular{T},::Type{Val{realmatrix}})
     if realmatrix
         TT = typeof(sqrt(zero(T)))
     else
-        TT = typeof(sqrt(complex(-one(T))))
+        TT = typeof(sqrt(complex(zero(T))))
     end
-    n = checksquare(A)
+    n = Base.LinAlg.checksquare(A)
     R = zeros(TT, n, n)
```

Comment:
My inclination would be to use `R = Matrix{TT}(n, n)` here.  Then get rid of the `r == 0` check below.  `sqrtm` is unlikely to return a sparse result, so I don't see the point of pre-initializing the array to zero.